spark3/src/main/java/org/apache/iceberg/spark/source/SparkMergeScan.java

```diff
@@ -39,31 +38,17 @@
 import org.apache.iceberg.io.FileIO;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
-import org.apache.iceberg.spark.Spark3Util;
-import org.apache.iceberg.spark.SparkReadOptions;
-import org.apache.iceberg.util.PropertyUtil;
 import org.apache.iceberg.util.TableScanUtil;
 import org.apache.spark.broadcast.Broadcast;
 import org.apache.spark.sql.connector.iceberg.read.SupportsFileFilter;
 import org.apache.spark.sql.connector.read.Statistics;
 import org.apache.spark.sql.util.CaseInsensitiveStringMap;
 
-import static org.apache.iceberg.TableProperties.SPLIT_LOOKBACK;
-import static org.apache.iceberg.TableProperties.SPLIT_LOOKBACK_DEFAULT;
-import static org.apache.iceberg.TableProperties.SPLIT_OPEN_FILE_COST;
-import static org.apache.iceberg.TableProperties.SPLIT_OPEN_FILE_COST_DEFAULT;
-import static org.apache.iceberg.TableProperties.SPLIT_SIZE;
-import static org.apache.iceberg.TableProperties.SPLIT_SIZE_DEFAULT;
-
 class SparkMergeScan extends SparkBatchScan implements SupportsFileFilter {
```

Comment:
I'm not sure if we should apply the same logic to `SparkMergeScan`. cc @aokolnychyi  for input on this.

Reply:
I think we should update the logic here to respect the by-partition setting. The merge-based queries could easily be used in joins when the operation has a subquery that is rewritten. Ideally, those joins should also support the bucketing strategy.